hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/NodeLauncher.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.mesos.hdfs.scheduler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.OfferID;
+import org.apache.mesos.SchedulerDriver;
+
+/**
+ * NodeLauncher. 
+ */
+public class NodeLauncher {
```

Comment:
This is not a large amount of logic and can be part of the HdfsNode class for clarity since this logic will be common to all HdfsNodes. We may want to rename this method attemptLaunch or tryLaunch if it makes sense.